voltdbclient/serialization_helper.go

```diff
@@ -0,0 +1,310 @@
+		return
+	}
+	serializedParams, err := serializeParams(params)
+	if err != nil {
+		return
+	}
+	io.Copy(&msg, &serializedParams)
+	return
+}
+
+func serializeParams(params []interface{}) (msg bytes.Buffer, err error) {
+	// parameter_count short
+	// (type byte, parameter)*
+	if err = writeShort(&msg, int16(len(params))); err != nil {
+		return
+	}
+	for _, val := range params {
+		if err = marshalParam(&msg, val); err != nil {
+			return
+		}
+	}
+	return
+}
+
+func marshalParam(buf io.Writer, param interface{}) (err error) {
+	v := reflect.ValueOf(param)
+	if !v.IsValid() {
+		return errors.New("Can not encode value.")
+	}
+	switch v.Kind() {
+	case reflect.Bool:
+		x := v.Bool()
+		writeByte(buf, vt_BOOL)
+		err = writeBoolean(buf, x)
+	case reflect.Int8:
+		x := v.Int()
+		writeByte(buf, vt_BOOL)
+		err = writeByte(buf, int8(x))
+	case reflect.Int16:
+		x := v.Int()
+		writeByte(buf, vt_SHORT)
+		err = writeShort(buf, int16(x))
+	case reflect.Int32:
+		x := v.Int()
+		writeByte(buf, vt_INT)
+		err = writeInt(buf, int32(x))
+	case reflect.Int, reflect.Int64:
+		x := v.Int()
+		writeByte(buf, vt_LONG)
+		err = writeLong(buf, int64(x))
+	case reflect.Float64:
+		x := v.Float()
+		writeByte(buf, vt_FLOAT)
+		err = writeFloat(buf, float64(x))
+	case reflect.String:
+		x := v.String()
+		writeByte(buf, vt_STRING)
+		err = writeString(buf, x)
+	case reflect.Struct:
+		if t, ok := v.Interface().(time.Time); ok {
+			writeByte(buf, vt_TIMESTAMP)
+			writeTimestamp(buf, t)
+		} else {
+			panic("Can't marshal struct-type parameters")
+		}
+	default:
+		panic(fmt.Sprintf("Can't marshal %v-type parameters", v.Kind()))
+	}
+	return
+}
+
+// readCallResponse reads a stored procedure invocation response.
+func deserializeCallResponse(r io.Reader) (response *Response, err error) {
+	response = new(Response)
+	if response.clientData, err = readLong(r); err != nil {
+		return nil, err
+	}
+
+	fields, err := readByte(r)
+	if err != nil {
+		return nil, err
+	} else {
+		response.fieldsPresent = uint8(fields)
+	}
+
+	if response.status, err = readByte(r); err != nil {
+		return nil, err
+	}
+	if response.fieldsPresent&(1<<5) != 0 {
+		if response.statusString, err = readString(r); err != nil {
+			return nil, err
+		}
+	}
+	if response.appStatus, err = readByte(r); err != nil {
+		return nil, err
+	}
+	if response.fieldsPresent&(1<<7) != 0 {
+		if response.appStatusString, err = readString(r); err != nil {
+			return nil, err
+		}
+	}
+	if response.clusterLatency, err = readInt(r); err != nil {
+		return nil, err
+	}
+	if response.fieldsPresent&(1<<6) != 0 {
+		if response.exceptionLength, err = readInt(r); err != nil {
+			return nil, err
+		}
+		if response.exceptionLength > 0 {
+			// TODO: implement exception deserialization.
+			ignored := make([]byte, response.exceptionLength)
+			if _, err = io.ReadFull(r, ignored); err != nil {
+				return nil, err
+			}
+		}
+	}
+	if response.resultCount, err = readShort(r); err != nil {
+		return nil, err
+	}
+
```

Comment:
if resultCount == 0 - is that an error condition?  Currently will return a response without any tables....